core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala

```diff
@@ -18,10 +18,17 @@
 package org.apache.spark.rdd
 
 import org.scalatest.FunSuite
+import org.scalatest.Matchers._
 
 import org.apache.spark._
 
 class DoubleRDDSuite extends FunSuite with SharedSparkContext {
+  test("sum") {
+    sc.parallelize(Seq.empty[Double]).sum() should be(0.0 +- 0.0001)
```

Comment:
Experience thought me that exact equality with doubles is unreliable. Now I err on the safe side and always use inexact matchers in unit tests.